enrollment-server-onboarding/src/main/java/com/wultra/app/onboardingserver/impl/service/IdentityVerificationService.java

```diff
@@ -376,40 +352,85 @@ public void processDocumentVerificationResult(final OwnerId ownerId, final Ident
             // The timestampFinished parameter is not set yet, there may be other steps ahead
-            moveToPhaseAndStatus(idVerification, phase, ACCEPTED, ownerId);
+            if (allRequiredDocumentsChecked) {
+                // Move to DOCUMENT_VERIFICATION / ACCEPTED only in case all documents were checked
+                moveToPhaseAndStatus(idVerification, IdentityVerificationPhase.DOCUMENT_VERIFICATION, ACCEPTED, ownerId);
+            } else {
+                // Identity verification status is changed to DOCUMENT_UPLOAD / IN_PROGRESS to allow submission of additional documents
+                moveToDocumentUpload(ownerId, idVerification, IN_PROGRESS);
+            }
         } else {
-            docVerifications.stream()
-                    .filter(docVerification -> docVerification.getStatus() == DocumentStatus.FAILED)
-                    .findAny()
-                    .ifPresent(failed -> {
-                        idVerification.setErrorDetail(failed.getErrorDetail());
-                        idVerification.setTimestampFailed(now);
-                        idVerification.setErrorOrigin(ErrorOrigin.DOCUMENT_VERIFICATION);
-                        moveToPhaseAndStatus(idVerification, phase, FAILED, ownerId);
-                    });
-
-            docVerifications.stream()
-                    .filter(docVerification -> docVerification.getStatus() == DocumentStatus.REJECTED)
-                    .findAny()
-                    .ifPresent(failed -> {
-                        idVerification.setErrorDetail(failed.getRejectReason());
-                        idVerification.setErrorOrigin(ErrorOrigin.DOCUMENT_VERIFICATION);
-                        idVerification.setTimestampFinished(now);
-                        moveToPhaseAndStatus(idVerification, phase, REJECTED, ownerId);
-                    });
+            // Identity verification status is changed to DOCUMENT_UPLOAD / IN_PROGRESS to allow re-submission of failed documents
+            moveToDocumentUpload(ownerId, idVerification, IN_PROGRESS);
+            handleDocumentStatus(docVerificationsToProcess, idVerification, DocumentStatus.FAILED, ownerId);
+            handleDocumentStatus(docVerificationsToProcess, idVerification, DocumentStatus.REJECTED, ownerId);
+        }
+    }
+
+    private void handleDocumentStatus(
+            final List<DocumentVerificationEntity> docVerifications,
+            final IdentityVerificationEntity idVerification,
+            final DocumentStatus status,
+            final OwnerId ownerId) {
+
+        docVerifications.stream()
+                .filter(docVerification -> docVerification.getStatus() == status)
+                .findAny()
+                .ifPresent(docVerification -> {
+                    logger.debug("At least one document is {}, ID: {}, {}", status, docVerification.getId(), ownerId);
+                    idVerification.setErrorDetail(docVerification.getRejectReason());
```

Comment:
- [ ] `ORA-12899: value too large for column "POWERAUTH"."ES_IDENTITY_VERIFICATION"."ERROR_DETAIL" (actual: 566, maximum: 256)`